### PR TITLE
proposal: update ticdc-mq-protocol-cn

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
add a section about kafka partition sharding algorithm

### Release note
- none

### CODE REVIEW

File: docs/design/2020-02-24-ticdc-mq-protocol-cn.md [hunk 1]
```diff
@@ -127,7 +127,44 @@ Value:
 
 ## Kafka 分区策略
 
-partition 数量不限，各个 partition 之间对等。
+* Row Changed Event 根据一定的分区算法分发至不同的 partition
+* Resolved Event 和 DDL Event 广播至所有的 partition
+
+### Row Changed Event 划分 Partition 算法
```
Comment (amyangfei) [inline docs/design/2020-02-24-ticdc-mq-protocol-cn.md]: What about adding a part to summarize which partition algorithm is used in TiCDC now?

**Comment by leoppro** [comment]
@amyangfei @overvenus  PTAL

File: docs/design/2020-02-24-ticdc-mq-protocol-cn.md [hunk 2]
```diff
@@ -127,7 +127,70 @@ Value:
...
+
+使用 pk(not handle) or uk(not null) 的 value 计算 hash 分发到 partition
+
+使用限制：
+
+* 表内只能包含一个 pk or uk
+
+#### 基于 table 分发
+
+使用 schemaID 和 tableID 计算 hash 分发到 partition
```
Comment (amyangfei) [inline docs/design/2020-02-24-ticdc-mq-protocol-cn.md:154]: 实际实现用的是 schemaName 和 tableName

Comment (leoppro) [inline docs/design/2020-02-24-ticdc-mq-protocol-cn.md:154]:
```suggestion
使用 schemaID 和 tableID 计算 hash 分发到 partition

注：在当前的实现中，基于 table 分发使用的是 schemaName 与 tableName 的值计算 hash
```

File: docs/design/2020-02-24-ticdc-mq-protocol-cn.md [hunk 3]
```diff
@@ -127,7 +127,72 @@ Value:
...
+
+使用 commit_ts 取模分发到 partition
+
+使用限制：
+
+* 无法保证行内有序性和表内有序性
+
+#### 冲突检测分发
+
+检测 pk or uk 冲突，将不冲突的 row 写入不同的 partition。发生冲突时向下游写入用于对齐进度的 Event
```
Comment (leoppro) [inline docs/design/2020-02-24-ticdc-mq-protocol-cn.md:172]:
```suggestion
检测 pk or uk 冲突，将不冲突的 row 写入不同的 partition。发生冲突时向下游写入用于对齐进度的 Event

注：尚未实现这种分发方式
```

**Comment by overvenus** [comment]
@liuzix PTAL

**Comment by leoppro** [comment]
ping @liuzix

**Pull request merged**